internal/controller/azurevalidator_controller.go

```diff
@@ -52,11 +54,7 @@ func (r *AzureValidatorReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	validator := &v1alpha1.AzureValidator{}
 	if err := r.Get(ctx, req.NamespacedName, validator); err != nil {
-		// Ignore not-found errors, since they can't be fixed by an immediate requeue
-		if apierrs.IsNotFound(err) {
-			return ctrl.Result{}, nil
-		}
-		r.Log.Error(err, "failed to fetch AzureValidator")
+		r.Log.Error(err, "failed to fetch AzureValidator", "key", req)
```

Comment:
Please refer to the latest version of this code in the Reconcile method for the validator controllers and/or the AWS/Network/OCI plugin controllers. @ahmad-ibra pointed out that we should make the logging conditional, otherwise an error is logged every time a validator is deleted.

Reply:
Fixed by making it conditional the same way as in AWS plugin.